sonic-ycabled/ycable/ycable_utilities/y_cable_helper.py

```diff
@@ -327,14 +333,69 @@ def toggle_mux_tor_direction_and_update_read_side(state, logical_port_name, phys
     if read_side is None or read_side is port_instance.EEPROM_ERROR or read_side < 0:
         helper_logger.log_error(
             "Error: Could not get read side for toggle command from orchagent Y cable port {}".format(logical_port_name))
+        port_instance.mux_toggle_status = port_instance.MUX_TOGGLE_STATUS_NOT_INITIATED_OR_FINISHED
         return (-1, -1)
     if int(read_side) == 1 or int(read_side) == 2:
         (active_side, read_side) = toggle_mux_direction(physical_port, read_side, state)
         return (active_side, read_side)
     else:
         #should not happen
+        port_instance.mux_toggle_status = port_instance.MUX_TOGGLE_STATUS_NOT_INITIATED_OR_FINISHED
         return (-1,-1)
 
+"""
+def poll_active_side_after_toggle(logical_port_name, read_side, requested_state, time_switchover_start):
+    physical_port_list = logical_port_name_to_physical_port_list(
+        logical_port_name)
+        
+
+    if len(physical_port_list) == 1:
+
+        physical_port = physical_port_list[0]
+        if physical_port != 1:
+            return
+        port_instance = y_cable_port_instances.get(physical_port)
+
+        required_side = 0
+        if read_side == '1':
+            if requested_state == "active":
+                required_side = 1
+            if requested_state == "standby":
+                required_side = 2
+        elif read_side == '2':
+            if requested_state == "active":
+                required_side = 2
+            if requested_state == "standby":
+                required_side = 1
+        else:
+            return
+
+        time_start = time.time()
+        count = 0
+        while True:
+            count = count + 1
+            curr_active_side = port_instance.get_mux_direction()
+            time_now = time.time()
+            time_diff = time_now - time_start
+            if curr_active_side == required_side:
+                break
+            elif time_diff >= TOGGLE_TIMEOUT:
+                helper_logger.log_warning(
+                    "Error: Could not toggle the mux for port {} {} to torA write eeprom timeout time taken {}, count: {}".format(physical_port, logical_port_name, time_now-time_switchover_start, count))
+                return -1
+
+            time.sleep(0.001)
```

Comment:
Just out of curiosity, were we polling every 1ms before? when we did the switchover measurement?